core/src/main/scala/cats/syntax/apply.scala

```diff
@@ -40,6 +41,211 @@ private[syntax] trait ApplySyntaxBinCompat0 {
     new IfApplyOps[F](fa)
 }
 
+private[syntax] trait ApplySyntaxBinCompat1 {
+  implicit final def catsSyntaxApplyFABOps[F[_], A, B](fab: F[A => B]): ApplyFABOps[F, A, B] =
+    new ApplyFABOps[F, A, B](fab)
+
+  implicit final def catsSyntaxApply2Ops[F[_], A, B, C](ff: F[(A, B) => C]): Apply2Ops[F, A, B, C] =
+    new Apply2Ops[F, A, B, C](ff)
+
+  implicit final def catsSyntaxProductOps[F[_], A, B](fa: F[A]): ProductOps[F, A, B] =
+    new ProductOps[F, A, B](fa)
```

Comment:
Oh, I doubt that type inference would work out well in this case. An implicit method should only include types that are part its parameters. Otherwise the compiler may not have a clue how to infer them. I would suggest 
```suggestion
  implicit final def catsSyntaxProductOps[F[_], A](fa: F[A]): YouNameIt[F, A] =
    new YouNameIt[F, A](fa)
```
then in the `YouNameIt`:
```scala
final class YouNameIt[F[_], A](private val fa: F[A]) extends AnyVal {
  def productR[B](fb: F[B])(implicit F: Apply[F]): F[B] = F.productR(fa)(fb)
}
```
I.e. the `B` type is introduces when it is actually used.
And the same for the other implicit methods below.

Feel free to give a try to your versions of course, but it looks suspicious to me personally.

Reply:
And of course all the methods that require just `F[A]` can be grouped together. Only methods that require more specialized types like `F[A => B]` have to be extracted.

Reply:
Oh, I didn't consider that aspect, thank you for pointing it out.

Fix: 0ee124ee806b3739ea22c521fb8a260f6eaf16cf